salespersonTrackerREST/models.py

```diff
@@ -1,3 +1,109 @@
 from django.db import models
-
+from django.contrib.auth.models import User
 # Create your models here.
+#Using Custom Django User Model
+#Managers be created via the admin only 
+#Initially give managers same username and password give them access to reset password
+
+
+
+class Manager(models.Model):
+    user_ref=models.ForeignKey(User,on_delete=models.CASCADE)
+    Name=models.CharField(max_length=100)
+    Photo=models.ImageField(upload_to='managers')
+    Age=models.IntegerField()
+
+    def __str__(self):
+        return self.user_ref.username
+
+#Salesperson will also be created via the admin only 
+#give salesperson the same username,password provide a link to reset their passwords
+
+class Salesperson(models.Model):
+    User_ref=models.ForeignKey(User,on_delete=models.CASCADE)
+    Managed_By=models.ForeignKey(Manager,on_delete=models.CASCADE)
+    Name=models.CharField(max_length=100)
+    Photo=models.ImageField(upload_to='salesperson')
+    Age=models.IntegerField()
+    last_location_lat=models.FloatField()
+    last_location_long=models.FloatField()
+
+    def __str__(self):
+        return self.User_ref.username
+
+# Item_Group_Code:
+#Classify items into their respective domains,like for eg a Toshiba Tv and a Samsung Tv can be grouped togheter under one roof named TV 
+#Company_item_code:A company can produce many diffrent models of Tv,hence to classify these models under one Manufacturer Equipment
+#They will be diffrent from each other by their item_code(i.e Barcode)
+
+
+
+#Item model is more a warehouse,where the mangers can view these items and assign them to the salesperson
+class Item(models.Model):
+    Item_Code=models.IntegerField(primary_key=True)
+    Item_Group_Code=models.IntegerField()
+    Company_Item_code=models.IntegerField()
+    Company_Code=models.IntegerField()
+    
+    Photo=models.ImageField(upload_to='Item')
+    Description=models.TextField()
+
+    def __int__(self):
+        return self.Item_Code
+    
+
+
+class ItemAssign(models.Model):
+    Item_Ref=models.ForeignKey(Item,on_delete=models.CASCADE)
+    Assigned_By=models.ForeignKey(Manager,on_delete=models.CASCADE)#Doubt about cascade
```

Comment:
Good to go with cascade as you will remove items assigned to the salesman if the manager is deleted.

Reply:
models.SET_NULL attribute added, where ever applicable